crates/core/src/sql/value/value.rs

```diff
@@ -167,2568 +172,344 @@ impl Ord for Value {
 	}
 
-	/// Try to convert this value to ab `Array` of a certain type
-	pub(crate) fn convert_to_array_type(self, kind: &Kind) -> Result<Array, Error> {
-		self.convert_to_array()?
-			.into_iter()
-			.map(|value| value.convert_to(kind))
-			.collect::<Result<Array, Error>>()
-			.map_err(|e| match e {
-				Error::ConvertTo {
-					from,
-					..
-				} => Error::ConvertTo {
-					from,
-					into: format!("array<{kind}>"),
-				},
-				e => e,
-			})
+	/// Converts a `surrealdb::sq::Value` into a `serde_json::Value`
+	///
+	/// This converts certain types like `Thing` into their simpler formats
+	/// instead of the format used internally by SurrealDB.
+	pub fn into_json(self) -> Json {
+		self.into()
 	}
 
-	/// Try to convert this value to ab `Array` of a certain type and length
-	pub(crate) fn convert_to_array_type_len(self, kind: &Kind, len: &u64) -> Result<Array, Error> {
-		self.convert_to_array()?
-			.into_iter()
-			.map(|value| value.convert_to(kind))
-			.collect::<Result<Array, Error>>()
-			.map_err(|e| match e {
-				Error::ConvertTo {
-					from,
-					..
-				} => Error::ConvertTo {
-					from,
-					into: format!("array<{kind}, {len}>"),
-				},
-				e => e,
-			})
-			.and_then(|v| match v.len() {
-				v if v > *len as usize => Err(Error::LengthInvalid {
-					kind: format!("array<{kind}, {len}>"),
-					size: v,
-				}),
-				_ => Ok(v),
-			})
-	}
+	// -----------------------------------
+	// Simple conversion of values
+	// -----------------------------------
 
-	/// Try to convert this value to an `Array` of a certain type, unique values
-	pub(crate) fn convert_to_set_type(self, kind: &Kind) -> Result<Array, Error> {
-		self.convert_to_array()?
-			.uniq()
-			.into_iter()
-			.map(|value| value.convert_to(kind))
-			.collect::<Result<Array, Error>>()
-			.map_err(|e| match e {
-				Error::ConvertTo {
-					from,
-					..
-				} => Error::ConvertTo {
-					from,
-					into: format!("set<{kind}>"),
-				},
-				e => e,
-			})
+	/// Treat a string as a table name
+	pub fn could_be_table(self) -> Value {
+		match self {
+			Value::Strand(v) => Value::Table(v.0.into()),
+			_ => self,
+		}
 	}
 
-	/// Try to convert this value to an `Array` of a certain type, unique values, and length
-	pub(crate) fn convert_to_set_type_len(self, kind: &Kind, len: &u64) -> Result<Array, Error> {
-		self.convert_to_array()?
-			.uniq()
-			.into_iter()
-			.map(|value| value.convert_to(kind))
-			.collect::<Result<Array, Error>>()
-			.map_err(|e| match e {
-				Error::ConvertTo {
-					from,
-					..
-				} => Error::ConvertTo {
-					from,
-					into: format!("set<{kind}, {len}>"),
-				},
-				e => e,
-			})
-			.and_then(|v| match v.len() {
-				v if v > *len as usize => Err(Error::LengthInvalid {
-					kind: format!("set<{kind}, {len}>"),
-					size: v,
-				}),
-				_ => Ok(v),
-			})
+	// -----------------------------------
+	// Simple output of value type
+	// -----------------------------------
+
+	/// Treat a string as a table name
```

Comment:
This comment is not right.

Reply:
This comment was there previously as this function was just moved but I will adjust it.

Reply:
Done